YoutubeDownloader/App.xaml.cs

```diff
@@ -21,37 +33,103 @@ public partial class App
     public static string ChangelogUrl { get; } = ProjectUrl + "/blob/master/Changelog.md";
 }
 
-public partial class App
+[DoNotNotify]
+public partial class App : StyletApplication<RootViewModel>
```

Comment:
because it seems like these were merged in Stylet.Avalonia https://github-com.translate.goog/sealoyal2018/Stylet.Avalonia?_x_tr_sl=auto&_x_tr_tl=de&_x_tr_hl=de

Reply:
Do you think we need Stylet at all? I remember we discussed this and you said Avalonia already provides many of its features out of the box (such as method binding). Does Avalonia have something for binding views to view models automatically too?

Reply:
I had started first without Stylet because it was still lacking behind in Avalonia previews when Avalonia 11 was still in preview. But since somethings changed with Avalonia till the release and also YoutubeDownloader moved forward I somewhat started again to not miss any of the changes. When I started again I used Stylet.Avalonia because it was faster. But I can again migrate of it. A drawback of the Stylet.Avalonia (which is a independet fork of Stylet) is also that is completey switched its readme to Chinese only.

Yes Avalonia has some parts already build in like binding to methods form commands (not from events) and one can easily build a general ViewLocator which is provided with the MVVM templates. I can change to that approach together with CommunityToolkit.MVVM from microsoft for some MVVM stuff. But I ended up rebuild some parts of Stylet, it would not be a problem for me since I had done that already.
You can also look at my first attempt here: https://github.dev/Mrxx99/YoutubeDownloader/tree/Avalonia/YoutubeDownloader.Avaloniav11

We could also wait until Caliburn.Micro is released (officially) for Avalonia which should be (hopefully) soon: https://github.com/Caliburn-Micro/Caliburn.Micro/pull/851

Reply:
> Yes Avalonia has some parts already build in like binding to methods form commands (not from events) and one can easily build a general ViewLocator which is provided with the MVVM templates. I can change to that approach together with CommunityToolkit.MVVM from microsoft for some MVVM stuff. But I ended up rebuild some parts of Stylet, it would not be a problem for me since I had done that already.

I'm pretty ambivalent, I think. We already have some "plumbing code" in `ViewModels/Framework` I think, so it wouldn't be too big of a deal if we added more. I'm also not opposed to adding 3rd party packages, as long as they're fairly widely used and are trustworthy.

I'll leave the decision up to you, let's go with whatever you think provides the best balance in terms of effort/control.

Reply:
> one can easily build a general ViewLocator which is provided with the MVVM templates. I can change to that approach together with CommunityToolkit.MVVM

if you're thinking of using the `CommunityToolkit.MVVM` tools, they also built a [DependencyInjection package](https://github.com/CommunityToolkit/Labs-Windows/discussions/463) with source generators that:
- Makes the service registration completely trim/AOT-friendly
- Provides built-time diagnostics on registration errors

With that, it's possible to build an improved version of the `ViewLocator`. [Here's](https://github.com/stevemonaco/AvaloniaViewModelFirstDemos) a demo project from @stevemonaco that uses the DI package.

The package is experimental, but apparently it is used in production and they're just waiting for the first chance to push it into the stable Nuget feed.

Reply:
> The package is experimental, but apparently it is used in production and they're just waiting for the first chance to push it into the stable Nuget feed.

I asked Sergio about this in the past week or so. He said that his MEDI sourcegen is currently being used in Windows Store, but he's lacking the time to move the code base/packaging into `CommunityToolkit` proper.